sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala

```diff
@@ -436,9 +436,8 @@ class DateExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     withSQLConf((SQLConf.ANSI_ENABLED.key, "true")) {
       checkErrorInExpression[SparkIllegalArgumentException](
         DateAddInterval(Literal(d), Literal(new CalendarInterval(1, 1, 25 * MICROS_PER_HOUR))),
-        "_LEGACY_ERROR_TEMP_2000",
-        Map("message" ->
-          "Cannot add hours, minutes or seconds, milliseconds, microseconds to a date",
+        "INVALID_PARAMETER_VALUE.INTERVAL_WITH_MICROSECONDS",
+        Map("parameter" -> "`interval`", "functionName" -> "`dateaddinterval`",
```

Comment:
`dateaddinterval ` there is no such function. That might confuse users. We could pass `+` instead, or create a error condition especially for the `DateAddInterval` expression (not sub-class of `INVALID_PARAMETER_VALUE `).